### ISSUE TITLE
E3701 false positive: Fn::If for a CodePipeline OutputArtifacts Name flagged as duplicate output-artifact name

### ISSUE THREAD

**Issue by whoDoneItAgain** [aws-cloudformation/cfn-lint#4584]
### CloudFormation Lint Version

cfn-lint 1.53.1

### What operating system are you using?

Windows

### Describe the bug

When an `AWS::CodePipeline::Pipeline` source action sets its `OutputArtifacts` `Name` with an `Fn::If`, cfn-lint reports **E3701 `'<name>' is already a defined 'OutputArtifact' Name`** for *both* branches of the `Fn::If`, even though only one artifact name is ever realized in a rendered template. The two branch names are mutually exclusive under the same condition, so no template ever defines the same output artifact name twice.

This comes up naturally when conditionally using the [full clone](https://docs.aws.amazon.com/codepipeline/latest/userguide/tutorials-github-gitclone.html) output format. A `CODEBUILD_CLONE_REF` artifact may only be consumed by CodeBuild actions, so when a parameter toggles the source between full clone and `CODE_ZIP`, the output artifact is given a different `Name` per mode (via `Fn::If`) so that non-CodeBuild consumers only ever bind to the `CODE_ZIP` artifact. cfn-lint treats the two conditional names as two simultaneous definitions on a single action and flags each as a duplicate of the other.

### Expected behavior

No E3701 is raised. In any rendered template exactly one `Fn::If` branch is realized, so the source action produces a single, uniquely-named output artifact. cfn-lint should resolve the condition scenarios (as it does for other rules) rather than treating the two mutually-exclusive branch names as duplicate definitions.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: >-
  Minimal reproduction of a cfn-lint E3701 false positive. A CodePipeline source
  action whose OutputArtifacts Name is a Fn::If is reported as defining a duplicate
  OutputArtifact name, even though exactly one artifact name is produced in any
  rendered template (the two Fn::If branches are mutually exclusive).

Parameters:
  UseFullClone:
    Type: String
    Default: 'false'
    AllowedValues:
      - 'true'
      - 'false'

Conditions:
  IsFullClone: !Equals [!Ref UseFullClone, 'true']

Resources:
  Pipeline:
    Type: AWS::CodePipeline::Pipeline
    Properties:
      RoleArn: !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:role/example-pipeline-role'
      ArtifactStore:
        Type: S3
        Location: example-artifact-bucket
      Stages:
        - Name: Source
          Actions:
            - Name: Source
              ActionTypeId:
                Category: Source
                Owner: AWS
                Provider: CodeStarSourceConnection
                Version: '1'
              Configuration:
                ConnectionArn: !Sub 'arn:${AWS::Partition}:codeconnections:${AWS::Region}:${AWS::AccountId}:connection/11111111-1111-1111-1111-111111111111'
                FullRepositoryId: example-org/example-repo
                BranchName: main
                OutputArtifactFormat: !If [IsFullClone, CODEBUILD_CLONE_REF, CODE_ZIP]
              OutputArtifacts:
                - Name: !If [IsFullClone, SourceGit, SourceZip]
              RunOrder: 1
        - Name: Build
          Actions:
            - Name: Build
              ActionTypeId:
                Category: Build
                Owner: AWS
                Provider: CodeBuild
                Version: '1'
              Configuration:
                ProjectName: example-build-project
              InputArtifacts:
                - Name: !If [IsFullClone, SourceGit, SourceZip]
              RunOrder: 1
```

### Command and output

```
> cfn-lint template.yaml
E3701 'SourceGit' is already a defined 'OutputArtifact' Name
template.yaml:46:43
E3701 'SourceZip' is already a defined 'OutputArtifact' Name
template.yaml:46:54
```

### Notes

Possibly related to (but distinct from) #4001, which is also E3701 but for a different scenario (multiple pipeline resources defined in one template). This report is specifically about a single conditional `OutputArtifacts` `Name` (`Fn::If`) on one action.
